huggingface_auth.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import base64
+import os
+from datetime import datetime, timedelta
+from getpass import getpass
+
+import requests
+from huggingface_hub import HfApi
+
+from hivemind.proto.auth_pb2 import AccessToken
+from hivemind.utils.auth import TokenAuthorizerBase
+from hivemind.utils.crypto import RSAPublicKey
+from hivemind.utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class HuggingFaceAuthorizer(TokenAuthorizerBase):
+    _AUTH_SERVER_URL = 'https://collaborative-training-auth.huggingface.co'
+
+    def __init__(self, experiment_id: int, username: str, password: str):
+        super().__init__()
+
+        self._experiment_id = experiment_id
+        self._username = username
+        self._password = password
```

Comment:
Same question as above about the need to store this information :slightly_smiling_face: .